frontend/src/routes/add_product/+page.svelte

```diff
@@ -0,0 +1,279 @@
+<script lang="ts">
+	import error from '$lib/assets/error.png';
+	import { Input } from '$lib/components/ui/input';
+	import transhcan from '$lib/assets/transhcan.png';
+
+	let data = [{ id: 0, category: '', subcategories: [1, 2, 3] }];
+
+	//id 辨識唯一data
+
+	let nextCategoryId = data.length + 1;
+
+	function addNewCategory() {
+		data = [
+			...data,
+			{ id: nextCategoryId, category: `New Category ${nextCategoryId}`, subcategories: [] }
+		];
+		nextCategoryId++;
+	}
+
+	function addNewSubcategory(index: number) {
+		const currentData = data[index];
+		data[index] = {
+			...currentData,
+			subcategories: [...currentData.subcategories, currentData.subcategories.length + 1]
+		};
+	}
+
+	function removeCategory(id: number) {
+		data = data.filter((cat) => cat.id !== id);
+	}
+
+	function removeNewSubcategory(index: number) {
+		const currentData = data[index];
+
+		// 如果 subcategories 不為空，則移除最後一個元素
+		if (currentData.subcategories.length > 0) {
+			currentData.subcategories.pop();
+		}
+
+		// 更新 data
+		data[index] = { ...currentData };
+	}
+
+	function addDiscountButton() {
+		buttons = [...buttons, { id: 6, label: `買二送一` }];
+	}
+
+	let buttons = [
+		{ id: 1, label: '買二送一' },
+		{ id: 2, label: '買二送一' },
+		{ id: 3, label: '買二送一' },
+		{ id: 4, label: '買二送一' },
+		{ id: 5, label: '買二送一' }
+	];
+
+	let count = 0;
+
+	function decrease() {
+		count--;
+	}
+
+	function increase() {
+		count++;
+	}
```

Comment:
components have counter to do this

Reply:
fix, change and use original function in counter.svelte